pkg/asset/installconfig/gcp/validation.go

```diff
@@ -113,6 +115,26 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 	return allErrs
 }
 
+// validateManagedZones will validate the public and private managed zones if they exist.
+func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.PublicDNSZone != "" && !strings.EqualFold(ic.GCP.PublicDNSZone, gcp.DNSZoneDisabled) {
+		if _, err := client.GetDNSZoneByName(context.TODO(), ic.GCP.ProjectID, ic.GCP.PublicDNSZone); err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("publicDNSZone"), ic.GCP.PublicDNSZone, "invalid public managed zone"))
+		}
+	}
+
+	if ic.GCP.PrivateDNSZone != "" && !strings.EqualFold(ic.GCP.PrivateDNSZone, gcp.DNSZoneDisabled) {
+		if _, err := client.GetDNSZoneByName(context.TODO(), ic.GCP.ProjectID, ic.GCP.PrivateDNSZone); err != nil {
```

Comment:
Couldn't this DNS zone be in a different project other than ic.GCP.ProjectID?

Reply:
Same applies as the public zone